x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/find_rules_status_route.ts

```diff
@@ -57,33 +46,43 @@ export const createFindRulesStatusRoute = (getClients: GetScopedClients): Hapi.S
             "anotherAlertId": ...
         }
     */
-    const statuses = await query.ids.reduce<Promise<RuleStatusResponse | {}>>(async (acc, id) => {
-      const lastFiveErrorsForId = await savedObjectsClient.find<
-        IRuleSavedAttributesSavedObjectAttributes
-      >({
-        type: ruleStatusSavedObjectType,
-        perPage: 6,
-        sortField: 'statusDate',
-        sortOrder: 'desc',
-        search: id,
-        searchFields: ['alertId'],
-      });
-      const accumulated = await acc;
-      const currentStatus = convertToSnakeCase<IRuleStatusAttributes>(
-        lastFiveErrorsForId.saved_objects[0]?.attributes
-      );
-      const failures = lastFiveErrorsForId.saved_objects
-        .slice(1)
-        .map(errorItem => convertToSnakeCase<IRuleStatusAttributes>(errorItem.attributes));
-      return {
-        ...accumulated,
-        [id]: {
-          current_status: currentStatus,
-          failures,
-        },
-      };
-    }, Promise.resolve<RuleStatusResponse>({}));
-    return statuses;
+    try {
+      const statuses = await query.ids.reduce<Promise<RuleStatusResponse | {}>>(async (acc, id) => {
+        const lastFiveErrorsForId = await savedObjectsClient.find<
+          IRuleSavedAttributesSavedObjectAttributes
+        >({
+          type: ruleStatusSavedObjectType,
+          perPage: 6,
+          sortField: 'statusDate',
+          sortOrder: 'desc',
+          search: id,
+          searchFields: ['alertId'],
+        });
+        const accumulated = await acc;
+        const currentStatus = convertToSnakeCase<IRuleStatusAttributes>(
+          lastFiveErrorsForId.saved_objects[0]?.attributes
+        );
+        const failures = lastFiveErrorsForId.saved_objects
+          .slice(1)
+          .map(errorItem => convertToSnakeCase<IRuleStatusAttributes>(errorItem.attributes));
+        return {
+          ...accumulated,
+          [id]: {
+            current_status: currentStatus,
+            failures,
+          },
+        };
+      }, Promise.resolve<RuleStatusResponse>({}));
+      return statuses;
+    } catch (err) {
+      const error = transformError(err);
+      return headers
+        .response({
+          message: error.message,
+          status_code: error.statusCode,
+        })
+        .code(error.statusCode);
+    }
```

Comment:
I see, I see.... This is all wrapped in a try / catch for safety. 👍 

This is good.